packages/legacy/src/client/helpers.ts

```diff
@@ -167,6 +181,14 @@ export const ovirtUrlToHostname = (url: string): string => {
 export const ovirtHostnameToUrl = (hostname: string): string =>
   `https://${hostname}/ovirt-engine/api`;
 
+export const openstackUrlToHostname = (url: string): string => {
+  const match = url.match(/^http:\/\/(.+)\:5000\/v3$/);
+  return match ? match[1] : url;
+};
+
+export const openstackHostnameToUrl = (hostname: string): string =>
+  `http://${hostname}:5000\/v3`;
```

Comment:
```suggestion
  `https://${hostname}:5000\/v3`;
```
?
a. http doesn't sound good
b. why are you changing the user input ?
if user say it's `https:/moki.com:999/my/openstack/v3.2/special/path`  we should respect that

Reply:
```suggestion
   // check if match full URL - e.g. match  http[s]://(hostname)(:port)?(/path)?
   // if match URL return ${hostname}
   
   // check if match (hostname) - e.g. no spaces no [:/] 
   // if match hostname return `http://${hostname}:5000\/v3`;
   
   if not a URL or hostname raise error
  
```

Reply:
Correct, the openStack url can consist of customized port and path parts and the API version can be v2 instead of v3 as well for old versions. Therefore as mentioned above, it seems better to display a URL field (with an example). 
A combination of url or hostname seems confusing to me.